src/Http/Routing/src/Builder/EndpointRoutingApplicationBuilderExtensions.cs

```diff
@@ -44,8 +45,18 @@ public static IApplicationBuilder UseRouting(this IApplicationBuilder builder)
 
             VerifyRoutingServicesAreRegistered(builder);
 
-            var endpointRouteBuilder = new DefaultEndpointRouteBuilder(builder);
-            builder.Properties[EndpointRouteBuilder] = endpointRouteBuilder;
+            IEndpointRouteBuilder endpointRouteBuilder;
+            if (builder.Properties.TryGetValue(GlobalEndpointRouteBuilderKey, out var obj))
+            {
+                endpointRouteBuilder = (IEndpointRouteBuilder)obj!;
+                // Let interested parties know if UseRouting() was called while a global route builder was set
+                builder.Properties["__UseRoutingWithGlobalSet"] = true;
```

Comment:
I was thinking about this and wondering if we should just set the existing `__EndpointRouteBuilder` key, wouldn't that just work in all cases?

The `__GlobalEndpointRouteBuilder` key indicates that we want to use the global endpoint route builder, while the other key says that we currently have an active endpoint route builder.

Reply:
Probably would work. It makes the code both cleaner and uglier, cleaner because we have one less new property, uglier because we're taking advantage of an existing property for something other than it's intended purpose.

Reply:
I think the intended purpose is the same as it was previously no?